astrocut/tests/test_FITSCutout.py

```diff
@@ -0,0 +1,537 @@
+    assert isinstance(cutout, str)
+
+    # Open output file
+    cutout_hdulist = fits.open(cutout)
+    assert len(cutout_hdulist) == len(test_images) + 1  # num imgs + primary header
+
+    # Check shape of data
+    cut1 = cutout_hdulist[1].data
+    assert cut1.shape == (cutout_size, cutout_size)
+    assert cutout_hdulist[1].data.shape == cutout_hdulist[2].data.shape
+    assert cutout_hdulist[2].data.shape == cutout_hdulist[3].data.shape
+    assert cutout_hdulist[3].data.shape == cutout_hdulist[4].data.shape
+    assert cutout_hdulist[4].data.shape == cutout_hdulist[5].data.shape
+    assert cutout_hdulist[5].data.shape == cutout_hdulist[6].data.shape
+
+    # Check that data is equal between cutout and original image
+    for i, img in enumerate(test_images):
+        with fits.open(img) as test_hdu:
+            assert np.all(cutout_hdulist[i + 1].data == test_hdu[0].data[19:29, 19:29])
+    
+    # Check WCS and position of center
+    cut_wcs = wcs.WCS(cutout_hdulist[1].header)
+    sra, sdec = cut_wcs.all_pix2world(cutout_size/2, cutout_size/2, 0)
+    assert round(float(sra), 4) == round(center_coord.ra.deg, 4)
+    assert round(float(sdec), 4) == round(center_coord.dec.deg, 4)
+
+    cutout_hdulist.close()
+
+
+def test_fits_cutout_multiple_files(tmpdir, test_images, center_coord, cutout_size):
+    # Output is multiple files
+    cutout_files = FITSCutout(test_images, center_coord, cutout_size, single_outfile=False, output_dir=tmpdir).cutout()
+
+    # Output is a list with paths to each file
+    assert isinstance(cutout_files, list)
+    assert len(cutout_files) == len(test_images)
+    
+    for i, cutout in enumerate(cutout_files):
+        cutout_hdulist = fits.open(cutout)
+        cut1 = cutout_hdulist[1].data
+
+        # Check shape of data
+        assert len(cutout_hdulist) == 2  # primary header + 1 image
+        assert cut1.shape == (cutout_size, cutout_size)
+        
+        # Check that data is equal between cutout and original image
+        with fits.open(test_images[i]) as test_hdu:
+            assert np.all(cut1 == test_hdu[0].data[19:29, 19:29])
+    
+        # Check WCS and position of center
+        cut_wcs = wcs.WCS(cutout_hdulist[1].header)
+        sra, sdec = cut_wcs.all_pix2world(cutout_size/2, cutout_size/2, 0)
+        assert round(float(sra), 4) == round(center_coord.ra.deg, 4)
+        assert round(float(sdec), 4) == round(center_coord.dec.deg, 4)
+
+    # Test case where output directory does not exist
+    new_dir = path.join(tmpdir, 'cutout_files')  # non-existing directory to write files to
+    cutout_files = FITSCutout(test_images, center_coord, cutout_size,
+                              output_dir=new_dir, single_outfile=False).cutout()
+
+    assert isinstance(cutout_files, list)
+    assert len(cutout_files) == len(test_images)
+    assert new_dir in cutout_files[0]
+    assert path.exists(new_dir)  # new directory should now exist
+
+    cutout_hdulist.close()
```

Comment:
Since `cutout_hdulist` is being iterated over, should this be within the loop to ensure each instance gets closed properly?